test-avocado/setup.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2013 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+#set -x
+SCRIPT_DIR=`dirname "$(readlink -f "$0")"`
+COCKPIT_DIR=$(cd $SCRIPT_DIR;cd ..; echo `pwd`)
+BASE=`basename $SCRIPT_DIR`
+source $SCRIPT_DIR/lib/host_setup.sh
+source $SCRIPT_DIR/lib/create_test_machine.sh
+LUSER=`whoami`
+ENV_VARIABLES=$COCKPIT_DIR/$BASE/lib/var.env
+rm -f $ENV_VARIABLES
+
+if check_host $LUSER; then
+    echo "Host already configured"
+else
+    if sudo $SCRIPT_DIR/lib/host_setup.sh $LUSER; then
+        echo "Host environemnt ready, please REBOOT you machine (to solve issue with default network changes)"
+        exit 0
+    else
+        echo "Unable to create enviromnet, EXITTING"
+        exit 1
+    fi
+fi
+
+# generate key for actual user if not generated yet (to ensure)
+if [ ! -e ~/.ssh/id_rsa ]; then
+    ssh-keygen -q -f ~/.ssh/id_rsa -N "" </dev/null
```

Comment:
Yes,
this part could be removed. It is only workaround becuase after creation of some user, keypairs are not generate. It is there only to ensure that key exist. 
It is not done silently, it is done vice-versa, don't do anything and ask in case key is already configured  (there is </dev/null, no  >/dev/null )